src/enhance.rs

```diff
@@ -0,0 +1,95 @@
+use crate::image::Image;
+use anyhow::Result;
+
+/// Performs weighted addition of two images `src1` and `src2` with weights `alpha`
+/// and `beta`, and an optional scalar `gamma`. The formula used is:
+///
+/// dst(x,y,c) = (src1(x,y,c) * alpha + src2(x,y,c) * beta + gamma)
+///
+/// # Arguments
+///
+/// * `src1` - The first input image.
+/// * `alpha` - Weight of the first image elements to be multiplied.
+/// * `src2` - The second input image.
+/// * `beta` - Weight of the second image elements to be multiplied.
+/// * `gamma` - Scalar added to each sum.
+///
+/// # Returns
+///
+/// Returns a new `Image` where each element is computed as described above.
+///
+/// # Errors
+///
+/// Returns an error if the sizes of `src1` and `src2` do not match.
+///
+/// # Panics
+///
+/// This function will panic if the sizes of `src1` and `src2` do not match. This is
+/// checked using `assert_eq!(src1.size(), src2.size())`.
+pub fn add_weighted<T, const CHANNELS: usize>(
+    src1: &Image<T, CHANNELS>,
+    alpha: &T,
+    src2: &Image<T, CHANNELS>,
+    beta: &T,
+    gamma: &T,
+) -> Result<Image<T, CHANNELS>>
+where
+    T: num_traits::Float + num_traits::FromPrimitive + std::fmt::Debug + Send + Sync + Copy,
+{
+    assert_eq!(src1.size(), src2.size());
```

Comment:
i return an error with anyhow for now -- im gonna put soon a patch handling custom errors